### ISSUE TITLE
mypy does not recognize SearchableAPIResource

### ISSUE THREAD

**Issue by jtetrault** [stripe/stripe-python#871]
### Describe the bug

When using `mypy` to type check code, an error is thrown when trying to access the `search` method of any resource that subclasses `SearchableAPIResource` (for example, `stripe.Price`).

Furthermore, VSCode's built-in Python language features also fail to recognize `search`.

![image](https://user-images.githubusercontent.com/3219018/188251127-f90a250c-a621-4122-a605-2ac10b229dcb.png)
![image](https://user-images.githubusercontent.com/3219018/188251166-70f485da-6334-4c5b-9a7a-630df3e5c5b3.png)

### To Reproduce

In an Ubuntu shell, run the following commands:
```bash
virtualenv venv
. venv/bin/activate
pip install stripe types-stripe mypy
echo "import stripe" > test.py
echo >> test.py
echo "stripe.Price.search" >> test.py
mypy test.py
```

`mypy` will output the following error:
```
test.py:3: error: "Type[Price]" has no attribute "search"
```

### Expected behavior

`mypy` should not throw an error here.

### Code snippets

```Python
import stripe

stripe.Price.search
```

### OS

Ubuntu 20.04.4 LTS on WSL2

### Language version

3.9.12

### Library version

stripe==4.1.0

### API version

2022-08-01

**Comment by yejia-stripe** [comment]
Hi @jtetrault, thanks for the report! This should be fixed with [types-stripe v3.5.2](https://github.com/typeshed-internal/stub_uploader/blob/main/data/changelogs/stripe.md#352-2022-09-13). I _believe_ VSCode's built-in Python language features also uses the typeshed stubs, but I'm not sure how those are updated.

Closing this out, but please feel free to re-open if this isn't resolved for you!

**Comment by jtetrault** [comment]
Awesome, thank you!
